tests/Fixtures/tags/inheritance/conditional_block_nested.legacy.test

```diff
@@ -0,0 +1,40 @@
+--TEST--
+conditional "block" tag with "extends" tag (nested)
+--DEPRECATION--
+Since twig/twig 3.14: Having a "block" tag under a "if" tag (line 7) is deprecated in layout.twig at line 8.
```

Comment:
This is a tag instead a non-top-level condition, so I don't think we need to deprecate this. Such tag is useful when you want to reuse part of the rendering between the `if` and `else` clause by defining the block in the `if` and referencing it in the `else`.
Having to move it to a top-level block (that does not override a parent block) makes the intent a lot less clear as it is easy to miss that this block is related to the rendering inside the condition (and so that it can be removed if the condition body does not need it anymore).

For the same reason, it can make sense to define a block inside the body of a loop, which would probably also trigger such a deprecation.

Reply:
To me, once entering a top-level block of a child template, we should not have any restriction on blocks anymore, just like there is no such restriction in a standalone template.